testscommon/chainSimulator/nodeHandlerMock.go

```diff
@@ -107,6 +110,14 @@ func (mock *NodeHandlerMock) GetStatusCoreComponents() factory.StatusCoreCompone
 	return nil
 }
 
+// GetIncomingHeaderSubscriber -
+func (mock *NodeHandlerMock) GetIncomingHeaderSubscriber() process.IncomingHeaderSubscriber {
```

Comment:
Should not be part of node handler

Reply:
We discussed about this. Had to make it accessible from here or from process components which required a lot more changes